### PR TITLE
Bugfix: mbstring function overloading has been removed in php 8

### PR DESCRIPTION
**Pull request by derrabus**
On php 8, the `MB_OVERLOAD_STRING` constant has been removed together with the mbstring function overload feature. In order to avoid an error when accessing the constant, I've saveguarded the access with a `defined()` call.

### CODE REVIEW

**Comment by derrabus** [comment]
Update: I've split the PHPUnit upgrade out to #120. Apparently, the test suite does run successfully with PHPUnit 7 on php 8, so that upgrade can be done independently (if you want to do it all).

File: .travis.yml [hunk 1]
```diff
@@ -47,17 +47,11 @@ matrix:
       dist: bionic
       env: USE_PSALM=0
            CHECK_MBSTRING=0
-    - php: "master"
```
Comment (paragonie-security) [inline .travis.yml:50]: Why is this removed?

Comment (derrabus) [reply .travis.yml:50]: What's the difference between `master` and `nightly`?

Comment (paragonie-security) [reply .travis.yml:50]: At one point, one was "latest nightly build of the new version" and the other was supposed to be "built from the master branch" (which in the near future should be called the `main` branch). But it looks like only `nightly` even exists right now.
